include/openmc/source.h

```diff
@@ -59,6 +59,15 @@ class SourceDistribution {
   UPtrDist energy_; //!< Energy distribution
 };
 
+class CustomSource {
+  public:
+    virtual ~CustomSource() {}
+
+    virtual Particle::Bank sample_source(uint64_t* seed) = 0;
```

Comment:
Since the type name includes "Source", it seems a little redundant to me to call this `sample_source`. I think `sample` is descriptive enough
```suggestion
    virtual Particle::Bank sample(uint64_t* seed) = 0;
```
At the point of use, it looks more natural:
```C++
custom_source->sample(seed)
```
rather than
```C++
custom_source->sample_source(seed)
```